action/lib/index.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
```

Comment:
This generated file should be ignored.

Reply:
It shouldn't be. This is the file that actually runs the action, as the actions have no way of compiling dependencies et. al.  Same for all below generated files.

Reply:
Are we sure there's no way to run tsc before running the action? This is generated code.

I'm sure there are examples of something like this:
https://github.com/jeffrafter/example-github-action-typescript

The workflow first builds then runs the action.

Reply:
Sure, but that pretty much negates the advantage I'm aiming for here (i.e. frameworks can just run the workflow). I think I can remove the lib/ file here, but the dist/ files should stay.